### DISCUSSION TITLE
windows.history.length is always 2, am I doing something wrong?

### DISCUSSION THREAD

**Discussion by nicholasfc** [vercel/next.js#16621] [Help]
I'm trying to get the window.history.length in order to show or not a back button with router.back(). My only problem is, if I open a new tab, visit the page directly via the url, my history is supposed to be 0, but I always get 2.

Am I doing something wrong and is window.history.length the best way to see if we have a history or not?

**Comment by gurupal** [comment]
did you able to fix this ? i am having the same issue

**Reply by nicholasfc** [reply]
Not really, ended up removing the back button 😅

**Comment by ungarida** [comment]
It is quite simple, create a link to that view and click on a new tab such link. In that case the `history.length` will be 1.

**Comment by sergey-hippo** [comment]
The issue is still present: if I open a URL from my Next.js app in a new tab, `history.length` is always 2

**Reply by sonyarianto** [reply]
when the new tab happen:
- open dev console and check history.length (now is 1)
- open your Next.js app path, let say home page /, check again dev console, now history.length = 2

**Reply by sonyarianto** [reply]
Is my solution valid like this?

```
if (window && window.history && window.history.length > 2) {
    window.history.back();
  } else {
    window.location.href = "/";
  }
```

**Reply by doong-jo** [reply]
I faced the problem the following situation.

- A page: route.push('/B');
- B page: route.back();
- A. page: console.log(window.history.length) // 2
